src/parameterizations/lateral/MOM_hor_visc.F90

```diff
@@ -504,7 +515,7 @@ real :: RoScl     ! The scaling function for MEKE source term [nondim]
     enddo ; enddo
 
     ! Components for the shearing strain
-    do J=js-2,Jeq+1 ; do I=is-2,Ieq+1
+    do J=js-2,Jeq+2 ; do I=is-2,Ieq+2
```

Comment:
@Hallberg-NOAA, thanks for the corrections. It is not clear to me what needs to be done to make this work properly with both symmetric or non-symmetric memory. Can you please provide additional guidance here?

Reply:
I would be happy to work through this in person, @gustavo-marques, but I also think that some general guidance on this issue might be more broadly useful.

The way to figure out these required loop extents is to start with the final output arrays (here diffu and diffv), and then work back through each of the intermediate steps to figure out the required array extents, bearing in mind that many of the intermediate arrays are used for more than one subsequent step, and also that `is-1 <= Isq <=  is` and `js-1 <= Jsq <= js' and `ie = Ieq` and `je = Jeq` . You can also quickly spot problematic extents by looking for cases where the number of extra halo points on each side of the computational domain should be the same with symmetric memory (for which `Isq = is-1`).  Also note that the loops need to be bigger with symmetric memory, so if it works in that case, it should also work with non-symmetric memory. The loops need to be big enough to calculate everything that is needed, but if they are taken to be too large it may lead to unnecessary halo updates.  One final test may be useful - if  large enough ranges are used, changing the values to which the arrays are initialized (or initializing them to NaNs) does not change the solutions.